### ISSUE TITLE
[no-bare-strings] Automatically allow all HTML entities

### ISSUE THREAD

**Issue by boris-petrov** [ember-template-lint/ember-template-lint#1131]
Things like `&times;` should be ignored by default by the `no-bare-strings` rule because they are built-in HTML constructs.

Forked from https://github.com/ember-template-lint/ember-template-lint/issues/662.

cc @rwjblue

**Comment by rwjblue** [comment]
Thank you @boris-petrov!

**Comment by rwjblue** [comment]
For context, I think that we can implement this by adding all known HTML entities to the default allow list. Thankfully, simple-html-tokenizer bundles the list of entities for us.

**Comment by Turbo87** [comment]
**why** should we allow any of these entities?

**Comment by lifeart** [comment]
@Turbo87 this entities seems multilingual (in most cases)

**Comment by Turbo87** [comment]
maybe some of them, but definitely not all. and I'm even doubtful about the rest. also, even if they are the same across all locales, why not still put them in the translations to be consistent?

**Comment by lifeart** [comment]
there is already list of `& * ;` items https://github.com/ember-template-lint/ember-template-lint/blob/master/lib/rules/no-bare-strings.js#L45

I think real-life use-cases for it it's pretty rare and we don't need to warn about it (because it can be meaningful inside app context - for example "calculator" app with special symbols)

**Comment by bmish** [comment]
I agree with having an option `allowHTMLEnttities` which defaults to true to maintain the current behavior.

**Comment by initram** [comment]
If we allow all HTML entities then we are not maintaining current behavior. Entities like &oslash; and &otilde; are not allowed by default today. I think it would weaken the rule to allow all HTML enties, especially if you look at the set of all HTML5 entities.

There is some merit to expanding the default white-list, but at some point we would have to tell people to write their own white-lists if they want more symbols allowed. If we allow &times; should we also allow the text "×"? Today we have a default whitelist that contains a bunch of commonly used symbols and their html entity equivalent.

**Comment by fusion2004** [comment]
@initram I think my main problem is I don't want to have to maintain the entire list. I just wanted to add one, and get the rest of default whitelist.

**Comment by rwjblue** [comment]
Ya, but we can work on that part regardless I think. A couple of ways:

* expose the default array for importing, so folks can do `[...defaults, 'thing-here']`
* add another option that is something like `additionalAllowedStrings` or something
